presto-spi/src/main/java/io/prestosql/spi/type/LongTimestampType.java

```diff
@@ -0,0 +1,147 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.type;
+
+import io.airlift.slice.XxHash64;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.block.BlockBuilderStatus;
+import io.prestosql.spi.block.Int96ArrayBlockBuilder;
+import io.prestosql.spi.block.PageBuilderStatus;
+import io.prestosql.spi.connector.ConnectorSession;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static io.prestosql.spi.type.UnscaledDecimal128Arithmetic.compare;
+
+public class LongTimestampType
+        extends TimestampType
+{
+    public LongTimestampType(int precision)
+    {
+        super(precision, LongTimestamp.class);
+    }
+
+    @Override
+    public int getFixedSize()
+    {
+        return Long.BYTES + Long.BYTES;
+    }
+
+    @Override
+    public BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries, int expectedBytesPerEntry)
+    {
+        int maxBlockSizeInBytes;
+        if (blockBuilderStatus == null) {
+            maxBlockSizeInBytes = PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+        }
+        else {
+            maxBlockSizeInBytes = blockBuilderStatus.getMaxPageSizeInBytes();
+        }
+        return new Int96ArrayBlockBuilder(
+                blockBuilderStatus,
+                Math.min(expectedEntries, maxBlockSizeInBytes / getFixedSize()));
+    }
+
+    @Override
+    public BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries)
+    {
+        return createBlockBuilder(blockBuilderStatus, expectedEntries, getFixedSize());
+    }
+
+    @Override
+    public BlockBuilder createFixedSizeBlockBuilder(int positionCount)
+    {
+        return new Int96ArrayBlockBuilder(null, positionCount);
+    }
+
+    @Override
+    public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        return compareTo(leftBlock, leftPosition, rightBlock, rightPosition) == 0;
+    }
+
+    @Override
+    public int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        long leftHigh = leftBlock.getLong(leftPosition, 0);
+        int leftLow = leftBlock.getInt(leftPosition, SIZE_OF_LONG);
+        long rightHigh = rightBlock.getLong(rightPosition, 0);
+        int rightLow = rightBlock.getInt(rightPosition, SIZE_OF_LONG);
+        return compare(leftLow, leftHigh, rightLow, rightHigh);
```

Comment:
I would not use the UnscaledDecimal128Arithmetic code for this.  Instead I would do something like
```
int value = Long.compare(leftHigh, rightHigh);
if (value != 0) {
    return value;
}
return Int.compareUnsigned(leftLow, rightLow);
```

Reply:
Good catch. That was not intended. I originally copied some methods from LongDecimalType and forgot to update.